signature/signature.go

```diff
@@ -0,0 +1,92 @@
+// Package signature provides functionality for interacting with the X-Flashbots-Signature header.
+package signature
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// HTTPHeader is the name of the X-Flashbots-Signature header.
+const HTTPHeader = "X-Flashbots-Signature"
+
+var (
+	ErrNoSignature      = errors.New("no signature provided")
+	ErrInvalidSignature = errors.New("invalid signature provided")
+)
+
+// Verify takes a X-Flashbots-Signature header and a body and verifies that the signature is valid for the body.
+// It returns the signing address if the signature is valid or an error if the signature is invalid.
+func Verify(header string, body []byte) (signingAddress string, err error) {
```

Comment:
I think the better API would be to return Address instead of the string because user has to wonder if its case sensitive and about 0x and Address is always Address.

WDYT?

Reply:
yup sgtm, I'll switch it to returning an Address

Reply:
Updated to return `common.Address` in 61c7374